src/index.js

```diff
@@ -1,57 +1,105 @@
+/* eslint-disable no-alert, no-plusplus, */
+
 import './index.css';
 
 /* Selectors */
 
-const items = JSON.parse(localStorage.getItem('items')) || [];
-const addBtn = document.querySelector("#add-btn");
+const todo = JSON.parse(localStorage.getItem('items')) || [];
 const list = document.querySelector('#list');
+const addBtn = document.querySelector('#add-btn');
+
+const storeItem = () => {
+  localStorage.setItem('items', JSON.stringify(todo));
+};
 
 /* To-do list displaying and storing */
 
 const todoList = () => {
-    items.sort((a, b) => a.index - b.index);
-  
-    items.forEach((item) => {
-        list.innerHTML += `<tr><td><input type="checkbox"><span class="item">${items.desc}</span></td></tr>`;
-    });
-};
+  todo.sort((a, b) => a.index - b.index);
 
-const storeItem = () => {
-    localStorage.setItem('items', JSON.stringify(items));
-} 
-  
-todoList();
+  todo.forEach((item) => {
+    list.innerHTML += `
+        <li> 
+            <input type="checkbox"><span class="item">${item.desc}<i class="fa fa-ellipsis-v"></i></span> 
+        </li>`;
+  });
+};
 
-/* To-do list functions */
+/* Add Item */
 
 const addItem = (desc) => {
-    const item = {
-        desc,
-        completed: false,
-        index: items.length + 1,
-    }
-    items.push(item);
-    storeItem();
-} 
-
-const removeTask = (index) => {
-    items.splice(index, 1);
-    for (let i = index; i < tasks.length; i += 1) {
-      items[i].index = i + 1;
+  const item = {
+    desc,
+    completed: false,
+    index: todo.length + 1,
+  };
+  todo.push(item);
+  storeItem();
+};
+
+addBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  const newItem = document.querySelector('#new').value;
+  if (!newItem) {
+    alert('Please add a task!');
+  } else {
+    addItem(newItem);
+    list.innerHTML += `
+            <li> 
+            <input type="checkbox"><span class="item">${newItem}<i id="edit" class="fa fa-ellipsis-v"></i></span> 
+            </li>`;
+    document.querySelector('#new').value = '';
+  }
+  storeItem();
+});
+
+/* Remove Item */
+
+const removeItem = (index) => {
+  todo.splice(index, 1);
+  for (let i = index; i < todo.length; i++) {
+    todo[i].index = i + 1;
+  }
+  storeItem();
+};
+
+/* Edit Item */
+
+const editItem = (index, desc) => {
+  todo[index].desc = desc;
+  storeItem();
+};
+
+/* Find Index */
+
+const findIndex = (e) => {
+  const items = document.querySelectorAll('.item');
+  let index = 0;
+  for (let i = 0; i < items.length; i++) {
+    const text = e.target.parentElement.textContent;
+    if (text === todo[i].desc) {
+      index = i;
```

Comment:
Please note that it is required to create a new JavaScript file for the new functionality.
![Screenshot from 2023-07-14 18-26-49](https://github.com/Arch-Noize/to-do-list/assets/44624138/e4334df3-2954-4edb-b137-040fcc8c0762)

Reply:
Oh I see, thank you for the reply. Do you mean I should separate my functions into a separate "Task" file and export it to index.js?

Reply:
- Well implemented